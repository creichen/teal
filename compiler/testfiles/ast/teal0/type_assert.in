fun f(x : int) = return (x : int);

fun f(x : array[int]) = return (x : array[any]);