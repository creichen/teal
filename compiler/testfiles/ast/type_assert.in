fun f(x : int) = return (x : int);

fun f(x : T[a, b, c]) = return (x : T [a]);