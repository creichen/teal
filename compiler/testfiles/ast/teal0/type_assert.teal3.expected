Module
  List
  List
    FunDecl
      IdDecl Identifier="f"
      Opt
      List
        VarDecl
          IdDecl Identifier="x"
          Opt
            Type
              IdUse Identifier="int"
              List
              List
          Opt
      Opt
        ReturnStmt
          TypedExpr
            Access
              IdUse Identifier="x"
            Type
              IdUse Identifier="int"
              List
              List
    FunDecl
      IdDecl Identifier="f"
      Opt
      List
        VarDecl
          IdDecl Identifier="x"
          Opt
            Type
              IdUse Identifier="array"
              List
                Type
                  IdUse Identifier="int"
                  List
                  List
              List
          Opt
      Opt
        ReturnStmt
          TypedExpr
            Access
              IdUse Identifier="x"
            Type
              IdUse Identifier="array"
              List
                Type
                  IdUse Identifier="any"
                  List
                  List
              List
